packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -474,29 +474,36 @@ export function getDevMiniflareOptions(
 				`Unexpected error: no specifier in request to module fallback service.`
 			);
 
-			const moduleRE = new RegExp(MODULE_PATTERN);
+			const moduleRE = new RegExp(ADDITIONAL_MODULE_PATTERN);
 			const match = moduleRE.exec(rawSpecifier);
 			assert(match, `Unexpected error: no match for module: ${rawSpecifier}.`);
 			const [full, moduleType, modulePath] = match;
+			assert(
+				moduleType,
+				`Unexpected error: module type not found in reference: ${full}.`
+			);
 			assert(
 				modulePath,
 				`Unexpected error: module path not found in reference: ${full}.`
 			);
 
-			let source: Buffer;
+			let contents: Buffer;
 
 			try {
-				source = fs.readFileSync(modulePath);
+				contents = fs.readFileSync(modulePath);
 			} catch (error) {
 				throw new Error(
 					`Import "${modulePath}" not found. Does the file exist?`
 				);
 			}
 
-			return MiniflareResponse.json({
-				// Cap'n Proto expects byte arrays for `:Data` typed fields from JSON
-				wasm: Array.from(source),
-			});
+			return MiniflareResponse.json(
+				{
+					CompiledWasm: { wasm: Array.from(contents) },
+					Data: { data: Array.from(contents) },
+					Text: { text: contents.toString() },
+				}[moduleType]
+			);
```

Comment:
It seems a bit unnecessary to generate each of these object just to throw away all but one of them. Can you use a switch instead?

```suggestion
			switch(moduleType) {
				case "CompiledWasm":
					return new MiniflareResponse.json({ wasm: Array.from(contents) });
				case "Data":
					return new MiniflareResponse.json({ data: Array.from(contents) });
				case "Text":
					return new MiniflareResponse.json({ text: contents.toString() });
				default:
					return new MiniflareResponse.error();
			)
```

Reply:
I would be amazed if this ever resulted in an observable difference but sure, I'll make the change. More significant, I think, is that I'm using the synchronous file system functions so I'll look at making the module fallback and `renderChunk` async.

Reply:
[83ec0bb](https://github.com/cloudflare/workers-sdk/pull/8387/commits/83ec0bb8f5240228472fe14b0fa0bcbc6ff4e892)
[64af99c](https://github.com/cloudflare/workers-sdk/pull/8387/commits/64af99cb945d2b4513770277917b07a1b291703a)

Reply:
Ignore my previous comment. You're right that it's wasteful because it's transforming the file contents. Didn't appreciate that difference before.